rest/revocation_test.go

```diff
@@ -1868,6 +1868,82 @@ func TestReplicatorRevocationsWithTombstoneResurrection(t *testing.T) {
 	assert.NoError(t, err)
 }
 
+func TestReplicatorRevocationsWithChannelFilter(t *testing.T) {
+	base.RequireNumTestBuckets(t, 2)
+
+	base.SetUpTestLogging(t, base.LevelDebug, base.KeyAll) // CBG-1981
+
+	// Passive
+	_, rt2 := InitScenario(t, nil)
+	defer rt2.Close()
+	rt2_collection := rt2.GetSingleTestDatabaseCollection()
+
+	// Active
+	rt1 := NewRestTesterDefaultCollection(t, //  CBG-2319: replicator currently requires default collection
+		&RestTesterConfig{
+			CustomTestBucket: base.GetTestBucket(t),
+		})
+	defer rt1.Close()
+	ctx1 := rt1.Context()
+
+	// Setup replicator
+	srv := httptest.NewServer(rt2.TestPublicHandler())
+	defer srv.Close()
+
+	passiveDBURL, err := url.Parse(srv.URL + "/db")
+	require.NoError(t, err)
+
+	passiveDBURL.User = url.UserPassword("user", "test")
+	sgwStats, err := base.SyncGatewayStats.NewDBStats(t.Name(), false, false, false, nil, nil)
+	require.NoError(t, err)
+	dbstats, err := sgwStats.DBReplicatorStats(t.Name())
+	require.NoError(t, err)
+
+	_ = rt2.CreateDocReturnRev(t, "docA", "", map[string][]string{"channels": []string{"ABC"}})
+	require.NoError(t, rt2.WaitForPendingChanges())
+
+	ar := db.NewActiveReplicator(ctx1, &db.ActiveReplicatorConfig{
+		ID:          t.Name(),
+		Direction:   db.ActiveReplicatorTypePull,
+		RemoteDBURL: passiveDBURL,
+		ActiveDB: &db.Database{
+			DatabaseContext: rt1.GetDatabase(),
+		},
+		Continuous:          false,
+		FilterChannels:      []string{"ABC"},
+		PurgeOnRemoval:      true,
+		ReplicationStatsMap: dbstats,
+	})
+
+	resp := rt2.SendAdminRequest("PUT", "/db/_user/user", GetUserPayload(t, "user", "test", "", rt2_collection, []string{"ABC"}, nil))
+	RequireStatus(t, resp, http.StatusOK)
+
+	require.NoError(t, ar.Start(ctx1))
+
+	defer func() {
+		assert.NoError(t, ar.Stop())
+	}()
+
+	// Wait for docs to turn up on local / rt1
+	changesResults, err := rt1.WaitForChanges(1, "/{{.keyspace}}/_changes?since=0", "", true)
+	require.NoError(t, err)
+	assert.Len(t, changesResults.Results, 1)
+
+	// Revoke A and ensure ABC chanel access and ensure DocA is purged from local
+	resp = rt2.SendAdminRequest("PUT", "/db/_user/user", GetUserPayload(t, "user", "test", "", rt2_collection, []string{}, nil))
+	RequireStatus(t, resp, http.StatusOK)
+
+	assert.NoError(t, ar.Stop())
```

Comment:
```suggestion
	require.NoError(t, ar.Stop())
```

If this step doesn't work the rest of the test doesn't work